src/stratis_cli/_actions/_top.py

```diff
@@ -204,9 +206,24 @@ def _add_update_key(proxy, key_desc, capture_key, *, keyfile_path):
     from ._data import Manager
 
     if capture_key:  # pragma: no cover
-        file_desc = sys.stdout.fileno()
-        fd_is_terminal = True
+        stdin_fd = sys.stdin.fileno()
+        (controller, controllee) = pty.openpty()
+        old_attrs = termios.tcgetattr(stdin_fd)
+        new_attrs = termios.tcgetattr(stdin_fd)
         print("Enter desired key data followed by the return key:")
+        new_attrs[3] = new_attrs[3] & ~termios.ECHO
+        termios.tcsetattr(stdin_fd, termios.TCSANOW, new_attrs)
+
+        # We can arbitrarily replace the newline with an ASCII control character
+        # that will cause the terminal to send EOF because readline() will only
+        # return a newline at the end of a buffered passphrase.
+        ascii_ctrl_d = chr(0x04)
+        pass_with_eof = sys.stdin.readline().replace("\n", ascii_ctrl_d)
```

Comment:
If you anyway read the passphrase in the stratis-cli process, why bother sending it to stratisd over a pseudo terminal? Couldn't that just be a normal pipe? Or even just a string argument in the D-Bus method call?

Reply:
@mvollmer While a pipe is an option, sending the passphrase over the D-Bus is not. It would be visible plain text to anyone monitoring the D-Bus. With regards to why I chose a pseudoterminal, this started when I still had terminal settings being set on the stratisd side. This required that the file descriptor being passed be a terminal or pseudoterminal. Now that I've removed that restriction, the pseudoterminal and pipe should theoretically work interchangeably. Would you prefer a pipe in this case?

Reply:
@mvollmer I've applied the change to a pipe and it works identically as expected.

Reply:
> While a pipe is an option, sending the passphrase over the D-Bus is not. It would be visible plain text to anyone monitoring the D-Bus. 

The system bus can only be monitored by root. Anybody else just sees the broadcasted signals. Or am I confused?

Reply:
> I've applied the change to a pipe and it works identically as expected.

Thanks! Any reduction in complexity is a win.